purchase_stock_secondary_unit/models/purchase_order_line.py

```diff
@@ -33,15 +33,18 @@ def write(self, vals):
         res = super().write(vals)
         for po_line in po_lines:
             moves = po_line.move_ids
+            secondary_uom_qty = vals["secondary_uom_qty"]
             if len(moves) == 1:
                 moves.filtered(lambda sm: sm.state not in ["done", "cancel"]).write(
-                    {"secondary_uom_qty": vals["secondary_uom_qty"]}
+                    {"secondary_uom_qty": secondary_uom_qty}
                 )
             elif moves and vals.get("secondary_uom_qty"):
                 previous_secondary_qty = sum(m.secondary_uom_qty for m in moves[:-1])
+                secondary_uom_qty = vals["secondary_uom_qty"] - previous_secondary_qty
                 moves[-1:].filtered(
                     lambda sm: sm.state not in ["done", "cancel"]
-                ).secondary_uom_qty = vals["secondary_uom_qty"] - previous_secondary_qty
+                ).secondary_uom_qty = secondary_uom_qty
+                moves[-1:].move_line_ids.write({"secondary_uom_qty": secondary_uom_qty})
```

Comment:
This is not correct, as you may have more than one move line per move and the secondary quantity should be the proportional one according the quantity in the sml.